tapir/coop/tests/membership_resignation/test_form.py

```diff
@@ -21,30 +26,102 @@ def setUp(self) -> None:
-        self.assertIn("transferring_shares_to", form.fields)
+    def test_membershipResignationForm_is_valid(self):
+        share_owner = ShareOwnerFactory.create()
+        resignation = MembershipResignationFactory.create()
+        data = {
+            "share_owner": share_owner,
+            "cancellation_reason": resignation.cancellation_reason,
+            "cancellation_date": resignation.cancellation_date,
+            "resignation_type": resignation.resignation_type,
+            "transferring_shares_to": resignation.transferring_shares_to,
+            "paid_out": resignation.paid_out,
+        }
+        form = MembershipResignationForm(data=data)
+        self.assertTrue(form.is_valid())
 
-    def test_validations(self):
-        self.assertRaises(
-            ValidationError, MembershipResignationForm.validate_duplicates
+    def test_validate_share_owner(self):
+        share_owner = ShareOwnerFactory.create()
+        resignation = MembershipResignationFactory.create(share_owner=share_owner)
+        form = MembershipResignationForm(data={"share_owner": resignation.share_owner})
+        form.validate_share_owner(resignation.share_owner)
+        self.assertIn("share_owner", form.errors.keys())
+        self.assertIn(
+            "This member is already resigned.",
+            form.errors["share_owner"],
+        )
+
+    def test_validate_transfer_choice(self):
+        share_owner = ShareOwnerFactory.create()
+        resignation = MembershipResignationFactory.create(
+            share_owner=share_owner,
+            resignation_type=MembershipResignation.ResignationType.TRANSFER,
+        )
+        form = MembershipResignationForm(
+            data={
+                "share_owner": resignation.share_owner,
+                "resignation_type": resignation.resignation_type,
+                "transferring_shares_to": resignation.transferring_shares_to,
+            }
+        )
+        form.validate_transfer_choice(
+            resignation.resignation_type,
+            transferring_shares_to=None,
         )
-        self.assertRaises(
-            ValidationError, MembershipResignationForm.validate_share_owner
+        self.assertIn("transferring_shares_to", form.errors.keys())
+        self.assertIn(
+            "Please select the member that the shares should be transferred to.",
+            form.errors["transferring_shares_to"],
         )
-        self.assertRaises(
-            ValidationError, MembershipResignationForm.validate_transfer_choice
+        form.validate_transfer_choice(
+            resignation_type=MembershipResignation.ResignationType.BUY_BACK,
+            transferring_shares_to=resignation.transferring_shares_to,
+        )
+        self.assertIn("transferring_shares_to", form.errors.keys())
+        self.assertIn(
+            "If the shares don't get transferred to another member, this field should be empty.",
+            form.errors["transferring_shares_to"],
```

Comment:
- This should be a separate test to make sure that there are no interferences between the two cases.
- There are two cases missing: ResignationType.TRANSFER with a valid member in `transferring_shares_to`, and `ResignationType.BUY_BACK` without member in  `transferring_shares_to`.
- We should check `is_valid` for each case too